api/types/ssh_roles.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// SSHRole identifies the role of an SSH connection. Unlike "user roles"
+// introduced as part of RBAC in Teleport 1.4+ these are built-in roles used
+// for different Teleport components when connecting to each other.
+type SSHRole string
+type SSHRoles []SSHRole
```

Comment:
Leave this named `Role`, they're not SSH specific.

Reply:
I've changed this to `TeleportRole` locally. the name `Role` would conflict with user `Role`.